### ISSUE TITLE
gpg key for apt.hyperion-project.org expired

### ISSUE THREAD

**Issue by trampi** [hyperion-project/hyperion.ng#1606]
Hi all,

when I try to run `apt-get update`, I get the following error:

```
Hit:1 http://deb.debian.org/debian bullseye InRelease
Get:2 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Hit:3 http://security.debian.org/debian-security bullseye-security InRelease
Hit:4 http://archive.raspberrypi.org/debian bullseye InRelease
Hit:5 https://apt.hyperion-project.org bullseye InRelease
Hit:6 https://download.docker.com/linux/debian bullseye InRelease
Err:5 https://apt.hyperion-project.org bullseye InRelease
  The following signatures were invalid: EXPKEYSIG 0E6710C4A760E210 Hyperion Project <admin@hyperion-project.org>
Fetched 44.1 kB in 3s (14.9 kB/s)
Reading package lists... Done
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://apt.hyperion-project.org bullseye InRelease: The following signatures were invalid: EXPKEYSIG 0E6710C4A760E210 Hyperion Project <admin@hyperion-project.org>
W: Failed to fetch https://apt.hyperion-project.org/dists/bullseye/InRelease  The following signatures were invalid: EXPKEYSIG 0E6710C4A760E210 Hyperion Project <admin@hyperion-project.org>
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

When I check the expiration date of the key, I see it expired May-08:
```
trampi@raspberrypi:~/homeassistant-docker $ gpg /usr/share/keyrings/hyperion.pub.gpg
gpg: WARNING: no command supplied.  Trying to guess what you mean ...
pub   rsa3072 2021-05-08 [SC] [expired: 2023-05-08]
      E858CEA18CAD56EB69AD83460E6710C4A760E210
uid           Paulchen-Panther <Paulchen-Panter@protonmail.com>
uid           Hyperion Project <admin@hyperion-project.org>
sub   rsa3072 2021-05-08 [E] [expired: 2023-05-08]
```

I also tried to re-download the key and either I failed doing that or the key is expired at the moment.

Best,
Fabian

**Comment by Paulchen-Panther** [comment]
Thank you for pointing this out. I will take care of the matter today.

**Comment by trampi** [comment]
Thanks! I appreciate all the hard work that all of the hyperion team (and especially you!) are putting into this.

**Comment by peteh** [comment]
I'm confused? Is this solved already? I still cannot update.

**Comment by Paulchen-Panther** [comment]
> I'm confused? Is this solved already? I still cannot update. 

We are in the process of fixing it. Please be patient

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT @ b2fcea3bbd5d5ee58ce56749cd9f98833664c8e4

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT @ b2fcea3bbd5d5ee58ce56749cd9f98833664c8e4

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT @ b2fcea3bbd5d5ee58ce56749cd9f98833664c8e4

**Comment by Paulchen-Panther** [comment]
@Hyperion-Bot build APT @ 24a00e3b0b9030b5d2cbf203b815397c737b25f9

**Comment by pattyland** [comment]
Maybe the issue could be reopened and closed when it's done? I was also searching the problem at my end, because this issue was closed. I read the comments because I didn't find anything

**Comment by 5w00sh** [comment]
> Maybe the issue could be reopened and closed when it's done? I was also searching the problem at my end, because this issue was closed. I read the comments because I didn't find anything

the following fixed my error:

wget -qO- https://apt.hyperion-project.org/hyperion.pub.key | sudo gpg --yes --dearmor -o /usr/share/keyrings/hyperion.pub.gpg && sudo apt-get updat

**Comment by maschbaer** [comment]
Same here,
I have got the same error on my bullseye as stated in the first post.

The proposed fix did not work for me.

Which hyperion apt-source should I use? So far I am on stable

Jan

**Comment by trampi** [comment]
@maschbaer for me the fix that 5w00sh posted worked.
The repository stays the same, just the key has been updated.

Maybe you can share your steps and the terminal output for further diagnosis / help? 😄

**Comment by Essy1245** [comment]
Today I ran:
wget -qO- https://apt.hyperion-project.org/hyperion.pub.key | sudo gpg --yes --dearmor -o /usr/share/keyrings/hyperion.pub.gpg

And got gpg: no valid OpenPGP data found. 
This command results in the same output. 
curl -fsSL https://releases.hyperion-project.org/hyperion.pub.key | sudo gpg --dearmor -o /usr/share/keyrings/hyperion.pub.gpg

At the end, I was trying to install this on a RPI3 running Raspberry Pi OS with Desktop Enviroment. 
As documentation suggests (Easy install script), I also tried running
curl -sSL https://releases.hyperion-project.org/install | bash -s -- --debian 'bookworm'
And got
bash: line 1: syntax error near unexpected token `newline'
bash: line 1: `<!DOCTYPE html>'
curl: (23) Failure writing output to destination

I first assumed something must be wrong my internet, but it appears to be fine. 
Then I went and checked the [documentation](https://docs.hyperion-project.org/user/gettingstarted/Linux.html), and I saw that it was updated 12 hours ago. 
The easy update script was updated yesterday. I've installed HyperionNG like two days ago on the same system with the same commands, I'm not sure what happened here.
